File: Java/qmk/_1002_写出这个数.java
import java.util.Scanner;

public class _1002_写出这个数 {
    public static void main(String[] arg){
        Scanner input = new Scanner(System.in);

        String number = input.next();
        int zz = Integer.valueOf(number).intValue();
        if(zz<Math.pow(10,100)) {


            int sum = 0;

            for (int i = 0; i < number.length(); i++) {
                sum += number.charAt(i) - '0';
            }


            int bai = sum / 100;   // num=398, 398/100=3.98, bai=3
            int shi = (sum % 100) / 10; // 398%100= 98, 98/10=9.8, shi=9
            int ge = ((sum % 100) % 10) / 1; // 98%10= 8


            String[] stringArray = {"ling", "yi", "er", "san", "si", "wu", "liu", "qi", "ba", "jiu"};

            if (bai != 0)
                System.out.print(stringArray[bai] + " ");
            if (shi != 0 ||  bai != 0)
                System.out.print(stringArray[shi] + " ");
            if (ge != 0 || shi != 0)
                System.out.print(stringArray[ge]);
        }
    }
}

File: C++/wdd/1002-写出这个数.cpp
#include <iostream>
#include <string>
using namespace std;

int main2()
{
    string n;
    cin >> n;
    string pinyin[] = { "ling", "yi", "er", "san", "si", "wu", "liu", "qi", "ba", "jiu" };

    int sum = 0;
    // string 转 int，求出 sum
    for (int i = 0; i < n.length(); i++) {
        n[i] -= '0';
        sum += n[i];
    }

    // int 转 string，方便循环，映射pinyin数组
    string sum_pinyin = to_string(sum);
    for (int i = 0; i < sum_pinyin.length(); i++) {
        cout << pinyin[sum_pinyin[i] - '0'];
        if (i < sum_pinyin.length() - 1) {
            cout << " ";
        }
    }
    cout << endl;

    return 0;
}